### PR TITLE
grpc-go client and server benchmark for protobuf

### PR DESCRIPTION
**Pull request by peter-edge**
This adds a grpc-go client to grpc-go server benchmark to the existing protobuf benchmarking. Right now we benchmark yarpc to yarpc http, tchannel, and thrift, and grpc-go client to yarpc-go server.

```
$ go test -bench Integration ./encoding/protobuf/testing
goos: darwin
goarch: amd64
pkg: go.uber.org/yarpc/encoding/protobuf/testing
BenchmarkIntegrationYARPC/http/Get-8         	   10000	    117196 ns/op
BenchmarkIntegrationYARPC/tchannel/Get-8     	   10000	    102670 ns/op
BenchmarkIntegrationYARPC/grpc/Get-8         	   10000	    148605 ns/op
BenchmarkIntegrationGRPCClient/Get-8         	   10000	    137745 ns/op
BenchmarkIntegrationGRPCAll/Get-8            	   10000	    104540 ns/op
PASS
ok  	go.uber.org/yarpc/encoding/protobuf/testing	6.220s
```

### CODE REVIEW

File: encoding/protobuf/testing/benchmark_test.go [hunk 1]
```diff
@@ -50,13 +52,29 @@ func BenchmarkIntegrationYARPC(b *testing.B) {
...
 }
 
+func BenchmarkIntegrationGRPCAll(b *testing.B) {
+	server := grpc.NewServer()
+	examplepb.RegisterKeyValueServer(server, example.NewKeyValueYARPCServer())
+	listener, err := net.Listen("tcp", "0.0.0.0:1234")
+	if err != nil {
+		b.Fatal(err.Error())
+	}
+	go func() { _ = server.Serve(listener) }()
```
Comment (akshayjshah) [inline encoding/protobuf/testing/benchmark_test.go]: Is there an error that we should check for and `b.Fatal` on?

Comment (peter-edge) [reply encoding/protobuf/testing/benchmark_test.go]: Per https://godoc.org/google.golang.org/grpc#Server.Serve, "Serve will return a non-nil error unless Stop or GracefulStop is called." We're technically calling stop, but then the semantics get messy to check everything, and checking for errors in the workflow isn't really the point of this benchmarking function.

Tldr: you generally ignore this error when doing simple tests/benchmarks like this

Comment (akshayjshah) [reply encoding/protobuf/testing/benchmark_test.go]: Fair enough, I buy it.
